promql/engine.go

```diff
@@ -2906,6 +2952,9 @@ func vectorElemBinop(op parser.ItemType, lhs, rhs float64, hlhs, hrhs *histogram
 			}
 		}
 	}
+	if parser.LTRIM == op || parser.RTRIM == op {
+		panic(fmt.Errorf("operator %q is only supported for native histograms and scalars, not standard vector operations", op))
```

Comment:
We probably shouldn't panic in these cases. There could be situations where this operation is done between a histogram and a vector that has a mix of floats and histograms. In such cases, the user might expect partial results instead of a full failure. It would be better to return an info annotation here rather than panicking.